### PR TITLE
Some little deriving cleanups

### PR DESCRIPTION
**Pull request by fitzgen**
See each commit for details.

r? @emilio

### CODE REVIEW

File: src/ir/analysis/derive_partial_eq_or_partial_ord.rs [hunk 1]
```diff
@@ -115,7 +124,7 @@ impl<'ctx> MonotoneFramework for CannotDerivePartialEqOrPartialOrd<'ctx> {
         trace!("constrain: {:?}", id);
 
         if self.cannot_derive_partialeq_or_partialord.contains(&id) {
-            trace!("    already know it cannot derive PartialEq or PartialOrd");
+            trace!("    already know it cannot derive `PartialEq`/`PartialOrd`");
```
Comment (pepyakin) [inline src/ir/analysis/derive_partial_eq_or_partial_ord.rs]: Is it makes sense to put markdownique syntax in logs?

File: src/ir/analysis/derive_partial_eq_or_partial_ord.rs [hunk 2]
```diff
@@ -13,31 +14,39 @@ use ir::ty::TypeKind;
...
 ///
 /// * If T is Opaque and layout of the type is known, get this layout as opaque
 ///   type and check whether it can be derived using trivial checks.
-/// * If T is Array type, partialeq or partialord cannot be derived if the length of
-///   the array is larger than the limit or the type of data the array contains cannot derive
-///   partialeq or partialord.
+///
+/// * If T is Array type, `PartialEq` or partialord cannot be derived if the
+///   length of the array is larger than the limit or the type of data the array
+///   contains cannot derive `PartialEq` or partialord.
```
Comment (pepyakin) [inline src/ir/analysis/derive_partial_eq_or_partial_ord.rs]: Is `partialord` intentional  or just missed?

Comment (fitzgen) [reply src/ir/analysis/derive_partial_eq_or_partial_ord.rs]: Just missed; thanks!

File: src/ir/analysis/derive_partial_eq_or_partial_ord.rs (hunk 1 above)
Comment (fitzgen) [reply src/ir/analysis/derive_partial_eq_or_partial_ord.rs]: I like to be consistent :)

**Pull request merged**